Add book index and show actions

storresy opened the pull request:
**Summary**
- Added Api V1 Controller to keep separated logic between API controllers and future back office controllers
- Added books controllers with list and show actions
- Added gem active_model_serializers
- Added Book serializer
- Added gem wor-paginate 
- Added pagination to book index

**Trello Card**
https://trello.com/c/MUnilUpy

**Sreenshots**
- Request without authentication
<img width="919" alt="screen shot 2018-12-11 at 11 20 56" src="https://user-images.githubusercontent.com/6606379/49815282-6c7e9b80-fd39-11e8-9730-b8c0ad1f0e1c.png">

- Authenticated request to list books end point
<img width="932" alt="screen shot 2018-12-11 at 11 20 12" src="https://user-images.githubusercontent.com/6606379/49815379-a9e32900-fd39-11e8-8856-ba22717603fe.png">
<img width="929" alt="screen shot 2018-12-11 at 11 20 31" src="https://user-images.githubusercontent.com/6606379/49815476-db5bf480-fd39-11e8-81c8-0c2888e5f53c.png">

- Authenticated request to book show
<img width="927" alt="screen shot 2018-12-11 at 11 20 38" src="https://user-images.githubusercontent.com/6606379/49815523-f595d280-fd39-11e8-886e-b972f083b536.png">

storresy committed:
Added book index and show actions

storresy committed:
wor paginate gem version fixed

Gmarn commented on app/controllers/api/v1/books_controller.rb:
```diff
@@ -0,0 +1,13 @@
+      end
+
+      def show
+        render json: Book.find(params[:id])
```
We use to require parameters to be sure frontend or external service send well information. Also you can define a method to do the find, like that:
```ruby
private

def book
  Book.find(params.require(:id))
end
```

2 more points: can you hanlde when the require raise an error o the find?

Gmarn commented on app/controllers/application_controller.rb:
```diff
@@ -1,4 +1,3 @@
 class ApplicationController < ActionController::Base
-  include DeviseTokenAuth::Concerns::SetUserByToken
   protect_from_forgery with: :null_session
```
Becarefull if you transfer the logic for Api un a new controller this should go in too.

Gmarn replied:
Nothing on this?

Gmarn commented on app/serializers/book_serializer.rb:
```diff
@@ -0,0 +1,3 @@
+class BookSerializer < ActiveModel::Serializer
+  attributes :id, :author, :title, :image, :editor, :year, :genre, :created_at, :updated_at
```
The doc ask for image field like this:
```json
    "image": {
      "url": null
    }
```

Gmarn replied:
For the `publisher` != `editor` do not mind

Gmarn commented on config/routes.rb:
```diff
@@ -2,7 +2,8 @@
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
-  api_version(:module => "api/1", :path => {:value => "api/v1"}) do
+  api_version(:module => "api/v1", :path => {:value => "api/v1"}) do
```
Becareful here you use old way to define hash.

storresy committed:
Fixed multiple sugestions made on PR

use new hash syntax on routes definition
Created new method on books controller to find a specific book
Rescue from not found and parameter missing on api v1 controller

storresy committed:
Fixed books controller error

Gmarn commented on app/controllers/api_v1_controller.rb:
```diff
@@ -0,0 +1,24 @@
+
+  def record_not_found
+    render json: {
+      'errors': [
```
Why passing a string here?
